Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -457,6 +464,24 @@ public final class PackageBuilder {
     /// Private function that creates and returns a list of targets defined by a package.
     private func constructTargets() throws -> [Target] {
 
+        // Ensure no dupicate target definitions are found
+        // Find duplicate targets
+        var uniqueTargets = Set<String>()
+        var duplicateTargets  = Array<String>()
+        for target in manifest.package.targets {
+            guard !uniqueTargets.contains(target.name) else {
+                duplicateTargets.append(target.name)
+                continue
+            }
+
+            uniqueTargets.insert(target.name)
+        }
```

Comment:
Can you factor this method into an extension or something? Seems like a useful thing to have.